spec/type/security_group_spec.rb

```diff
@@ -1,31 +1,65 @@
 require 'spec_helper'
+require 'awspec/error'
+
 Awspec::Stub.load 'security_group'
 
 describe security_group('sg-1a2b3cd4') do
   it { should exist }
-  its(:inbound) { should be_opened(80) }
+  it do
+    # only way found to force and test the exception
+    subject.inbound
+    expect { subject.opened?(nil) }.to raise_error(Awspec::MissingPortSpecification, /protocol\sis\s-1$/)
+  end
+  its(:inbound) { should_not be_opened(80) }
+  its(:inbound) { should_not be_opened(80).protocol(-1) }
+  its(:inbound) { should be_opened(80).protocol(-1).for('sg-3a4b5cd6') }
   its(:inbound) { should be_opened(80).protocol('tcp').for('123.45.68.89/32') }
   its(:inbound) { should be_opened(80).protocol('tcp').for('123.45.67.0/25') }
   its(:inbound) { should be_opened(80).protocol('tcp').for('123.45.67.1/32') }
   its(:inbound) { should_not be_opened(80).protocol('tcp').for('123.45.0.0/16') }
-  its(:inbound) { should be_opened(22) }
+
+  its(:inbound) { should be_opened(22).protocol('tcp').for('sg-1a2b3cd4') }
+  its(:inbound) { should be_opened(22).protocol('udp').for('sg-1a2b3cd4') }
+  its(:inbound) { should be_opened(22).protocol('-1').for('sg-1a2b3cd4') }
+  its(:inbound) { should be_opened(22).protocol('icmp').for('sg-1a2b3cd4') }
+
   its(:inbound) { should be_opened(22).protocol('tcp').for('sg-5a6b7cd8') }
   its(:inbound) { should be_opened('50000-50009').protocol('tcp').for('123.45.67.89/32') }
   its(:inbound) { should_not be_opened('50010-50019').protocol('tcp').for('123.45.67.89/32') }
-  its(:outbound) { should be_opened(50_000) }
+
+  its(:outbound) { should be_opened(50_000).protocol('tcp').for('sg-1a2b3cd4') }
+  its(:outbound) { should be_opened(50_000).protocol('udp').for('sg-1a2b3cd4') }
+  its(:outbound) { should be_opened(50_000).protocol('icmp').for('sg-1a2b3cd4') }
+  its(:outbound) { should be_opened(50_000).protocol('-1').for('sg-1a2b3cd4') }
+
   its(:outbound) { should be_opened(8080).protocol('tcp').for('sg-9a8b7c6d') }
   its(:outbound) { should be_opened(8080).protocol('tcp').for('group-in-other-aws-account-with-vpc-peering') }
   its(:inbound) { should be_opened_only(60_000).protocol('tcp').for('100.45.67.12/32') }
   its(:inbound) { should be_opened_only(70_000).protocol('tcp').for(['100.45.67.89/32', '100.45.67.12/32']) }
   its(:outbound) { should be_opened_only(50_000).protocol('tcp').for('100.45.67.12/32') }
-  its(:inbound) { should be_opened.protocol('all').for('sg-3a4b5cd6') }
```

Comment:
I'm concerned about the loss of the `all` test.
I agree with the inclusion of `-1`, but would like to see `all` supported for backward compatibility.